src/part_file.cpp

```diff
@@ -310,19 +315,27 @@ namespace libtorrent
 			{
 				int const slot = i->second;
 				open_file(file::read_only, ec);
+				boost::shared_ptr<file> local_file = m_file;
 				if (ec) return;
 
 				if (!buf) buf.reset(new char[m_piece_size]);
 
+				// don't hold the lock during disk I/O
+				l.unlock();
```

Comment:
I added these `unlock()` and `lock()` back again, as I believe they are safe now